photon-ml/src/test/scala/com/linkedin/photon/ml/optimization/GeneralizedLinearOptimizationProblemTest.scala

```diff
@@ -14,100 +14,215 @@
  */
 package com.linkedin.photon.ml.optimization
 
-import com.linkedin.photon.ml.data.{LabeledPoint, SimpleObjectProvider}
-import com.linkedin.photon.ml.function.TwiceDiffFunction
-import com.linkedin.photon.ml.model.Coefficients
-import com.linkedin.photon.ml.normalization.NormalizationContext
-import com.linkedin.photon.ml.sampler.DownSampler
-import com.linkedin.photon.ml.supervised.model.{GeneralizedLinearModel, ModelTracker}
-import com.linkedin.photon.ml.supervised.classification.LogisticRegressionModel
-import com.linkedin.photon.ml.test.CommonTestUtils
+import scala.math.abs
 
-import breeze.linalg.Vector
-import breeze.linalg.sum
-import org.apache.spark.rdd.RDD
-import org.mockito.Matchers
+import breeze.linalg.{Vector, sum}
 import org.mockito.Mockito._
 import org.testng.Assert._
 import org.testng.annotations.Test
 
-import scala.collection.mutable
-import scala.math.abs
+import com.linkedin.photon.ml.constants.MathConst.HIGH_PRECISION_TOLERANCE_THRESHOLD
+import com.linkedin.photon.ml.data.LabeledPoint
+import com.linkedin.photon.ml.function._
+import com.linkedin.photon.ml.function.svm.IndividualSmoothedHingeLossFunction
+import com.linkedin.photon.ml.model.Coefficients
+import com.linkedin.photon.ml.supervised.classification.{LogisticRegressionModel, SmoothedHingeLossLinearSVMModel}
+import com.linkedin.photon.ml.supervised.model.GeneralizedLinearModel
+import com.linkedin.photon.ml.supervised.regression.{LinearRegressionModel, PoissonRegressionModel}
+import com.linkedin.photon.ml.test.CommonTestUtils
 
+/**
+ * Test the base function in GeneralizedLinearOptimizationProblem common to all optimization problems.
+ */
 class GeneralizedLinearOptimizationProblemTest {
-  import GeneralizedLinearOptimizationProblemTest._
   import CommonTestUtils._
+  import GeneralizedLinearOptimizationProblemTest._
 
   @Test
-  def testRun(): Unit = {
```

Comment:
Moved to the IndividualOptimizationProblemTest and DistributedOptimizationProblemIntegTest.